### ISSUE TITLE
Is StyleGAN3 able to generate realistic samples for custom bedroom dataset?

### ISSUE THREAD

**Issue by MartiGrau** [NVlabs/stylegan3#166]
**Introduction of my problem:**
I'm trying to train a StyleGAN3 model with a custom bedroom dataset similar to the LSUN Bedroom dataset. But, I'm unable to achieve realistic results like the StyleGAN3 models trained for faces generation. (For me, realism == no artifacts, detailed objects, no strange patterns)

In the following image, you can see the realism differences between the official checkpoint of StyleGAN3 model for faces and my training for bedrooms at 512x512 resolution, both:

<img width="580" alt="image" src="https://user-images.githubusercontent.com/37666194/171826686-6a1fe5ff-dd41-4ec3-ae55-9e8d105667e9.png">

**Information of my datasets and tests that I've done:**
- Hardware: 4 GPUs RTX3090 24Gb of memory each.
- Dataset: 80K bedroom images with more than 1024x1024 resolution.
- Tests: I've tried to train for different resolutions (256x256, 512x512, 1024x1024) with different hyperparameters and for several days...
- The best FID obtained was 4,9 after 9 training days.

I've been reading other papers, some blogs and twitter threads related to the training of StyleGAN3 models, and I've tried to follow their advices without improvement...
- I tried to reduce the variability of my dataset trying to take only 'master' bedrooms + bedrooms with only 2 o 3 walls + all bedrooms with windows on the left side, flipping all the bedrooms with right window (Final dataset of 30k images)
- I've tried to start from an official pre-trained checkpoint of faces dataset with high realism`stylegan3-t-ffhqu-256x256.pkl` and freezing the discriminator layers at first 2000kimg with not much improvement (Train converging faster without realism of my generated bedrooms).
- I've tried different StyleGAN configurations ('stylegan2', 'stylegan3-t', 'stylegan3-r') without improvement.

**My questions:**
- What can I expect from the realism of StyleGAN for complex datasets like bedroom scenes? Is it impossible to obtain realism from 'ffhqu' dataset (faces)??
- Which hyperparameters do you recommend me to train this type of complex scenes?
- Is it enough 80k images to train a realistic bedroom generator?
- I've also read that stylegan2 architecture could be better for datasets like mine. What do you think about that?
- Why didn't the authors trained a LSUN Bedroom model for StyleGAN2 and StyleGAN3, but they did it for StyleGAN 1? Could this artefact be related to the difficulty of training this type of datasets?

Thank you in advance.

**Comment by PDillis** [comment]
I'd argue that `--cfg=stylegan2` should suit you best, as StyleGAN3 isn't meant to improve the image quality. The bedrooms dataset you have is a far harder dataset than FFHQ/FFHQU, as even FFHQ still has a face with its usual features (eyes, mouth, nose, head, etc.), whereas the bedroom can have so many differences (distance to each bed, different types/sizes of pillows, windows, etc.). 

If you really want to get the StyleGAN3There are ways to mitigate this. Currently, I'm running an experiment with ~800 images I collected from [r/roomporn](https://www.reddit.com/r/roomporn) (so your 80k images should be more than enough). Right now it's at 1680 kimg and it looks like so:

![image](https://user-images.githubusercontent.com/24496178/172026030-cf274405-9317-4020-8e00-58628ec6f69b.png)

The key here was setting `--gamma=100`, with `--cfg=stylegan3-t` and starting from the WikiArt found in Justin Pinkney's [Awesome Pretrained StyleGAN3 repo](https://github.com/justinpinkney/awesome-pretrained-stylegan3). You could use the Benches model, though it's at 512x512 resolution. In the end, a high gamma and batch size (using `--gpus 4 --batch=32 --batch-gpu=8`) will allow the model to better learn a parameter space, where afterward you can then use a lower learning rate, lower gamma, lower batch size, etc., to find a better local minimum. 

Given the number of images you have, you could also start from FFHQU or start from scratch, but perhaps it would be helpful if you share your previous experiments and the parameters you used.

**Comment by MartiGrau** [comment]
Thank you for your response @PDillis. 
I've finally decided to train my dataset using StyleGAN2-ADA instead of StyleGAN3 models and I'm getting better results. The high variability of this type of datasets is the key to train a GAN model with good quality.

In short, what I've done to improve quality for my bedroom dataset:

-  Use StyleGAN2-ADA architecture
- Create subsets of my dataset according to [manhattan world assumption](https://papers.nips.cc/paper/2000/file/90e1357833654983612fb05e3ec9148c-Paper.pdf) == number of walls, ceiling and floor in the bedroom image.
- I've fine-tuned the model from a 256x256 official checkpoint freezing the 3 first layers of the discriminator for 2000kimg

Here some of the results:
<img width="513" alt="image" src="https://user-images.githubusercontent.com/37666194/173192146-849aa87f-835e-445c-89c5-542994d6cd8b.png">
